shapeshift/web
# feat: multi account send and receive

## Description
Adds foundational support for sending and receiving from BIP44 accounts besides 0` for all supported chains.
Leverages accountMetadata on the portfolio in the redux store.

## Notice




## Pull Request Type


## Issue (if applicable)



closes #2612 
closes #2602
closes #2601 

## Risk


This affects the way we derive the parameters we send to chain-adapters and hdwallet when asking for a receive address and when asking for a send transaction signature for all supported chains as well as scriptTypes (account types).

The risk of deriving addresses at a wallet path we didn't intend to would/could lead to incompatibilities with other HD Wallets or even treasure hunts for missing funds (litecoin on bitcoin path etc).

## Testing
Test both send and receive for all supported chains+tokens plus account types (i.e. Legacy vs Segwit vs P2SH)
I like to use 3rd party wallets to sanity check wallet development. load the same seed in trust/ledger/trezor etc and verify the receive addresses/balances match this applications'.

For utxos it's important to spend new utxos to be thorough.
- get a BTC Segwit Receive Address
-  send some BTC to it
- verify trust wallet (or any other wallet) sees the new btc balance
- spend the new btc sending it where ever
- repeat for others in the chain/scriptType matrix


### Engineering


a for loop in AppContext works well to add accounts in lieu of scanning/add functionality.
### Operations



## Screenshots
![Screen Shot 2022-09-02 at 3 03 28 PM](https://user-images.githubusercontent.com/7322275/188965477-db243225-16f6-4cd5-be49-f03b81d863b7.png)
![Screen Shot 2022-09-02 at 3 00 33 PM](https://user-images.githubusercontent.com/7322275/188965610-b0607711-b4ef-4e71-aa4f-815730a056ce.png)
